### PR TITLE
Update unit testing guide

### PR DESCRIPTION
**Pull request by withinboredom**
The existing documentation doesn't work and is outdated. This is an attempt to update it with two different options for running the unit tests.

## Testing

Feel free to follow the directions from scratch and verify, if you run into any issues it may be that I skipped a step, or wasn't very clear; so let me know if something doesn't work...

### CODE REVIEW

File: docs/unit-testing.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+Jetpack includes several [unit tests](https://github.com/Automattic/jetpack/tree/master/tests) that you can run in your
+ local environment before to submit a new Pull Request.
```
Comment (singerb) [inline docs/unit-testing.md]: Should be 'before you submit', probably.

File: docs/unit-testing.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Prerequisites
+
+To get started, you can follow the instructions [here](https://phpunit.de/getting-started.html) to install PHPUnit on 
+your machine. Once you've done so, you can get the WordPress testing codebase like so:
+
```
Comment (singerb) [inline docs/unit-testing.md]: Missing the svn checkout line from the original file.

Comment (singerb) [inline docs/unit-testing.md]: Hmm, I see that line below actually. In that case, change or remove this wording.

File: docs/unit-testing.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+You'll also need to install a mySQL server, to run the tests.
+
+# Option 1: Everything inside Jetpack folder
```
Comment (singerb) [inline docs/unit-testing.md]: Maybe some comments about what the difference between these two methods is? Will they both run the same tests, it's just a matter of where you're running them from?

**Comment by singerb** [comment]
I'm happy with the wording changes, but someone more familiar with Jetpack testing than I am should probably verify the steps.

**Comment by eliorivero** [comment]
I personally don't think we should suggest users to download wp core into Jetpack plugin folder. In that way, they have to point the WP_DEVELOP_DIR into the directory inside Jetpack folder which might not be the best if they're developing and phpunit-testing several plugins at the same time.

The way my setup is done on macOS is like this:

1. an svn clone of wp core in `/Users/%username%/Sites/`
2. a production site inside, for example `/Users/%username%/Sites/automattic` and in turn Jetpack is in `/Users/%username%/Sites/automattic/wp-content/plugins/jetpack`
3. `WP_DEVELOP_DIR=/Users/%username%/Sites/tests/phpunit`

And that's it. I can always freely update the svn copy, and can run phpunit from within Jetpack folder or any other folder that has a `phpunit.xml` without having to clone wp src again.

**Comment by oskosk** [comment]
@withinboredom do you think this one still makes sense ?

**Comment by withinboredom** [comment]
Forgot all about this thing. I'm going to close it for now.

**Comment by oskosk** [comment]
Alright! fwiw I think this was a great addition all in all. If you eventually want to reopen and rebase, ping me for review!

**Comment by withinboredom** [comment]
I'll revisit in the new year for sure!

**Pull request closed without merging**